### PR TITLE
refac: deprecate `vitess/go/maps2` for `golang.org/x/exp/maps`

### PR DESCRIPTION
**Pull request by EshaanAgg**
## Description

Remove unneeded `maps2` package since `golang.org/x/exp/maps` provides these as well.

## Related Issue(s)

Fixes part of #14931 

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Review by dbussink — CHANGES_REQUESTED** [review]
I don't think this is the right change here. What we want is to remove `maps2` since these are functions that Go provides these days in `golang.org/x/exp/maps`.

**Comment by dbussink** [comment]
> I don't think this is the right change here. What we want is to remove `maps2` since these are functions that Go provides these days in `golang.org/x/exp/maps`.

@EshaanAgg Is this something you are interested in tackling? Otherwise I can clean it up as well.

File: go/vt/vttablet/tabletserver/schema/engine.go [hunk 1]
```diff
@@ -52,6 +50,8 @@ import (
 	"vitess.io/vitess/go/vt/vttablet/tabletserver/connpool"
 	"vitess.io/vitess/go/vt/vttablet/tabletserver/tabletenv"
 
+	"golang.org/x/exp/maps"
+
```
Comment (dbussink) [inline go/vt/vttablet/tabletserver/schema/engine.go:54]: This import should go above the `vitess.io` ones.

File: go/vt/wrangler/switcher_dry_run.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
 	"strings"
 	"time"
 
-	"vitess.io/vitess/go/maps2"
+	"golang.org/x/exp/maps"
```
Comment (dbussink) [inline go/vt/wrangler/switcher_dry_run.go:27]: This should have a  newline after this one before the `vitess.io` imports.

File: go/vt/vttablet/tabletserver/schema/engine.go (hunk 1 above)
Comment (EshaanAgg) [reply go/vt/vttablet/tabletserver/schema/engine.go:54]: Got it!

File: go/vt/wrangler/switcher_dry_run.go (hunk 2 above)
Comment (EshaanAgg) [reply go/vt/wrangler/switcher_dry_run.go:27]: Aight!

**Comment by EshaanAgg** [comment]
The code coverage workflow seems to fail, but I can't seem to figure out the reason for the same (can't see any changes in this PR that should affect those tests). Can someone help me debug that?

**Comment by ajm188** [comment]
likely a flaky test: https://github.com/vitessio/vitess/actions/runs/7543640140/job/20537054175?pr=14960#step:9:403

**Review by mdlayher — APPROVED** [review]

**Review by rohit-nayak-ps — APPROVED** [review]
Thanks for the contribution.

**Pull request merged**